### PR TITLE
fix(errors): do not log authjs message with CredentialsSiginin error

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- Change `CredentialsSignin` error to use `Error` as base class
	- Avoids adding our "Read more at authjs.dev .." suffix to `error.message` when users extend this class for their own signin errors

Replaces https://github.com/nextauthjs/next-auth/pull/10231

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Comment by superafroman** [comment]
@ndom91 this breaks the logic that redirects these errors to the signin page (see `isAuthError` in src/index.ts). Now a CredentialsSignIn error shows the error page with `error=Configuration`

**Comment by dml0031** [comment]
> @ndom91 this breaks the logic that redirects these errors to the signin page (see `isAuthError` in src/index.ts). Now a CredentialsSignIn error shows the error page with `error=Configuration`

Also noticed this, left some more details on this in this [comment](https://github.com/nextauthjs/next-auth/pull/10943#issuecomment-2143943734) on the other PR that added `code` to the client side.

**Comment by yukinoyuu** [comment]
Breaks the types on CredentialsSignin errors. The typing of CredentialsSignin is changed from extending SignInError to extending Error.

Screenshot 2024-06-14 at 11 12 13 AM

File: packages/core/src/errors.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export class InvalidCallbackUrl extends AuthError {
  * 1. The user is redirected to the signin page, with `error=CredentialsSignin&code=credentials` in the URL. `code` is configurable.
  * 2. If you throw this error in a framework that handles form actions server-side, this error is thrown, instead of redirecting the user, so you'll need to handle.
  */
-export class CredentialsSignin extends SignInError {
+export class CredentialsSignin extends Error {
```
Comment (yukinoyuu) [inline packages/core/src/errors.ts:196]: Breaks types on CredentialsSignin #11155

**Review by Kupinaaa — COMMENTED** [review]
Breaks types on CredentialsSignin class #11155

**Comment by l0gicgate** [comment]
This PR caused a regression and now we are unable to throw custom errors during the sign in mechanism.

Is there a plan to address this? I can raise a PR to revert if necessary.

@ndom91

**Comment by ndom91** [comment]
> This PR caused a regression and now we are unable to throw custom errors during the sign in mechanism.
> 
> Is there a plan to address this? I can raise a PR to revert if necessary.
> 
> @ndom91 

Thanks for the reminder. 

Check out my response here: https://github.com/nextauthjs/next-auth/issues/11155#issuecomment-2187149916

I'll take a look at fixing this shortly

**Comment by l0gicgate** [comment]
Thank you for the prompt response @ndom91!

**Pull request merged**